src/index.test.ts

```diff
@@ -157,6 +157,30 @@ describe('post()', () => {
     // assert error is empty
     expect(error).toBe(undefined);
   });
+
+  it('supports optional requestBody', async () => {
+    const mockData = { status: 'success' };
+    const client = createClient<paths>();
+    fetchMocker.mockResponse(() => ({ status: 201, body: JSON.stringify(mockData) }));
+
+    // assert omitting `body` doesn’t raise a TS error (testing the response isn’t necessary)
+    await client.post('/create-tag/{name}', {
+      params: { path: { name: 'New Tag' } },
+    });
+
+    // assert providing `body` with correct schema doesn’t raise a TS error
+    await client.post('/create-tag/{name}', {
+      params: { path: { name: 'New Tag' } },
+      body: { description: 'This is a new tag' },
+    });
+
+    // assert providing `body` with bad schema WILL raise a TS error
+    await client.post('/create-tag/{name}', {
+      params: { path: { name: 'New Tag' } },
+      // @ts-expect-error
```

Comment:
This is another great feature of TypeScript: this will actually throw an error if this **IS NOT** invalid TypeScript. Neat!